### ISSUE TITLE
Rule Change: change default behavior of no-unused-vars to warning instead of error

### ISSUE THREAD

**Issue by k98kurz** [eslint/eslint#19451]
### What rule do you want to change?

no-unused-vars

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
let anything = "something";
// do not use anything
```

### What does the rule currently do for this code?

Generates an error. Makes code editors turn red for valid syntax.

### What will the rule do after it's changed?

Generate a warning. Don't make everything think that valid syntax is an error.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

The experience of having valid syntax reported as an error is infuriating. Only invalid syntax should produce errors that make the code editor turn red. To write valid code with the default eslint configuration is to literally see red everywhere, all the time, and for no reason. To use a var, it must first be defined -- reporting the act of defining a new variable as an error is unhinged.

**Comment by nzakas** [comment]
If you prefer a warning, you can just set that in your config file:

```js
{
    rules: {
        "no-unused-vars": "warn"
    }
}
```

**Comment by k98kurz** [comment]
@nzakas So it is the consensus of the community that valid syntax is an error?

**Comment by k98kurz** [comment]
@nzakas Fundamentally, you did not address the issue. The issue is that an insane default behavior should be replaced with a sane default. Your response amounts to a requirement that everyone change the behavior of eslint in every project and repository they have to maintain to fix a bad default setting. What is the rationale for having a bad default setting?
